src/topology/order/scott.lean

```diff
@@ -0,0 +1,469 @@
+begin
+  intros h,
+  rw is_open_eq_upper_and_lub_mem_implies_tail_subset at h,
+  exact h.1,
+end
+
+lemma is_closed_is_lower {s : set (with_scott_topology α)} : is_closed s → is_lower_set s :=
+begin
+  intro h,
+  rw is_closed_eq_lower_and_subset_implies_lub_mem at h,
+  exact h.1,
+end
+
+/--
+The closure of a singleton `{a}` in the Scott topology is the right-closed left-infinite interval
+(-∞,a].
+-/
+@[simp] lemma closure_singleton (a : with_scott_topology α) : closure {a} = Iic a :=
+begin
+  rw ← lower_set.coe_Iic,
+  rw ← lower_closure_singleton,
+  refine subset_antisymm _ _,
+  { apply closure_minimal subset_lower_closure,
+    rw is_closed_eq_lower_and_subset_implies_lub_mem,
+    split,
+    { exact (lower_closure {a}).lower },
+    { rw lower_closure_singleton,
+      intros d b d₁ d₂ d₃ d₄,
+      rw [lower_set.coe_Iic, mem_Iic],
+      exact (is_lub_le_iff d₃).mpr d₄, } },
+  { apply lower_closure_min subset_closure (is_closed_is_lower _),
+    apply is_closed_closure, }
+end
+
+lemma continuous_monotone {f : (with_scott_topology α) → (with_scott_topology β)}
+  (hf: continuous f) : monotone f :=
+begin
+  rw monotone,
+  intros a b hab,
+  let u := (Iic (f b))ᶜ,
+  by_contra,
+  have u2 : a ∈ (f⁻¹'  u) := h,
+  have s1 : is_open u :=
+  begin
+    rw [is_open_compl_iff, ← closure_singleton],
+    exact is_closed_closure,
+  end,
+  have s2 :  is_open (f⁻¹'  u) := is_open.preimage hf s1,
+  have u3 : b ∈ (f⁻¹'  u) := is_upper_set_iff_forall_le.mp s2.1 hab u2,
+  have c1 : f(b) ∈ (Iic (f b))ᶜ :=
+  begin
+    simp only [mem_compl_iff, mem_preimage, mem_Iic, le_refl, not_true] at u3,
+    simp only [mem_compl_iff, mem_Iic, le_refl, not_true],
+    exact u3,
+  end,
+  simp only [mem_compl_iff, mem_Iic, le_refl, not_true] at c1,
+  exact c1,
+end
+
+end with_scott_topology
+
+lemma preserve_lub_on_directed_iff_scott_continuity
+  (f : (with_scott_topology α) → (with_scott_topology β)) :
+  preserve_lub_on_directed f ↔ continuous f :=
+begin
+  split,
+  { intro h,
+    rw continuous_def,
+    intros u hu,
+    rw with_scott_topology.is_open_eq_upper_and_lub_mem_implies_inter_nonempty,
+    split,
+    { apply is_upper_set.preimage (with_scott_topology.is_open_is_upper hu),
+      apply preserve_lub_on_directed_montotone,
+      exact h, },
+    { intros d a hd₁ hd₂ hd₃ ha,
+    have e1: is_lub (f '' d) (f(a)) :=
+    begin
+      apply h,
+      apply hd₁,
+      apply hd₂,
+      apply hd₃,
+    end,
+    rw with_scott_topology.is_open_eq_upper_and_lub_mem_implies_inter_nonempty at hu,
+    have e2: ((f '' d) ∩ u).nonempty :=
+    begin
+      apply hu.2,
+      exact nonempty.image f hd₁,
+      have e3: monotone f := begin
+        apply preserve_lub_on_directed_montotone,
+        exact h,
+      end,
+      apply directed_on_image.mpr,
+      exact directed_on.mono hd₂ e3,
+      apply e1,
+      exact ha,
+    end,
+    exact image_inter_nonempty_iff.mp e2, } },
```

Comment:
Indents are off here, the `have` line and everything below it isn't aligned with the `intros`